From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 23 Feb 2019 11:07:44 +0100
Subject: [PATCH] {BUG in 5.0-rc8} mm/shmem: initialize the return value

clang reports:

    mm/shmem.c:2860:6: error: variable 'ret' is used uninitialized
    whenever 'if' condition is false
    [-Werror,-Wsometimes-uninitialized]
            if (inode->i_nlink) {
                ^~~~~~~~~~~~~~
    mm/shmem.c:2873:9: note: uninitialized use occurs here
            return ret;
                   ^~~
    mm/shmem.c:2860:2: note: remove the 'if' if its condition is always
    true
            if (inode->i_nlink) {
            ^~~~~~~~~~~~~~~~~~~~
    mm/shmem.c:2851:9: note: initialize the variable 'ret' to silence
    this warning
            int ret;
                   ^
                    = 0
---
 mm/shmem.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/shmem.c b/mm/shmem.c
index 9b059fc95d20..6932f34c479a 100644
--- a/mm/shmem.c
+++ b/mm/shmem.c
@@ -2848,7 +2848,7 @@ static int shmem_create(struct inode *dir, struct dentry *dentry, umode_t mode,
 static int shmem_link(struct dentry *old_dentry, struct inode *dir, struct dentry *dentry)
 {
 	struct inode *inode = d_inode(old_dentry);
-	int ret;
+	int ret = 0; /* BUG */
 
 	/*
 	 * No ordinary (disk based) filesystem counts links as inodes;
-- 
